homeassistant/components/proxmox/sensor.py

```diff
@@ -0,0 +1,143 @@
+"""Support for sensors to show the resource usages of Proxmox VE."""
+import homeassistant.components.proxmox as proxmox
+from homeassistant.helpers.entity import Entity
+
+
+async def async_setup_platform(
+        hass, config, async_add_entities, discovery_info=None):
+    """Set up Proxmox VE sensors."""
+    nodes = hass.data[proxmox.DATA_PROXMOX_NODES]
+    sensors = []
+
+    for name in nodes.keys():
+        item = nodes[name]
+        if 'type' in item and item['type'] == 'node':
+            sensors.append(
+                PXMXSensor(
+                    hass, name, 'Uptime', uptime, "min", 'mdi:clock-outline'))
+        else:
+            sensors.append(
+                PXMXSensor(hass, name, 'VCPU Count', vcpus, None, 'mdi:chip'))
+        sensors.append(
+            PXMXSensor(hass, name, 'Memory', ram, "%", 'mdi:memory'))
+        sensors.append(
+            PXMXSensor(hass, name, 'Disk', disk, "%", 'mdi:harddisk'))
+        sensors.append(
+            PXMXSensor(hass, name, 'CPU', cpu, "%", 'mdi:chip'))
+        sensors.append(
+            PXMXSensor(
+                hass, name, 'Max. Memory', mem_max, 'GB', 'mdi:memory'))
+        sensors.append(
+            PXMXSensor(
+                hass, name, 'Memory Used', mem_used, 'GB', 'mdi:memory'))
+        sensors.append(
+            PXMXSensor(
+                hass, name, 'Max. Disk', disk_max, 'GB', 'mdi:harddisk'))
+        sensors.append(
+            PXMXSensor(
+                hass, name, 'Disk Used', disk_used, 'GB', 'mdi:harddisk'))
+        sensors.append(
+            PXMXSensor(hass, name, 'Status', status, None, None))
+
+    async_add_entities(sensors)
+
+
+class PXMXSensor(Entity):
+    """Sensors to show the resource usages of Proxmox VE nodes & VMs."""
+
+    def __init__(self, hass, node_name, sensor_name, value, unit, icon):
+        """Initialize Proxmox VE sensor."""
+        self._hass = hass
+        self._node_name = node_name
+        self._unit = unit
+        self._icon = icon
+        self._sensor_name = sensor_name
+        self._value = value
+        self._state = None
+        self._is_available = False
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return '{} ({})'.format(self._sensor_name, self._node_name)
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement of this entity, if any."""
+        return self._unit
+
+    @property
+    def available(self):
+        """Return True if Monitor is available."""
+        return self._is_available
+
+    @property
+    def icon(self):
+        """Return the icon to use in the frontend, if any."""
+        return self._icon
+
```

Comment:
I have the following suggestions:
- add a function for `unique_id`
- add the vmid in the `device_state_attributes`, because if the user changes the name of the entity, he can't directly see the vmid anymore